WordPress/Jetpack/Classes/ViewRelated/WordPress Migration/Welcome/MigrationWelcomeViewModel.swift

```diff
@@ -0,0 +1,70 @@
+import UIKit
+
+final class MigrationWelcomeViewModel: MigrationStepViewModel {
+
+    // MARK: - Properties
+
+    let blogListDataSource: BlogListDataSource
+
+    // MARK: - Init
+
+    init(account: WPAccount, primaryAction: @escaping () -> Void) {
+        self.blogListDataSource = BlogListDataSource()
+        self.blogListDataSource.loggedIn = true
+        self.blogListDataSource.account = account
+
+        let primaryAction = Action(title: Strings.primaryButtonTitle, handler: primaryAction)
+        let secondaryAction = Action(title: Strings.secondaryButtonTitle, handler: {})
+        let actions = Actions(primary: primaryAction, secondary: secondaryAction)
+
+        let secondaryDescription = Strings.secondaryDescription(plural: blogListDataSource.visibleBlogsCount > 1)
+        let descriptions = Descriptions(primary: Strings.primaryDescription, secondary: secondaryDescription)
+
+        super.init(
+            email: account.email,
+            title: Strings.title,
+            image: UIImage(named: "wp-migration-welcome"),
+            descriptions: descriptions,
+            actions: actions
+        )
+    }
+
+    // MARK: - Constants
+
+    struct Strings {
+
+        static let title = NSLocalizedString(
+            "migration.welcome.title",
+            value: "Welcome to Jetpack!",
+            comment: "The title in the migration welcome screen"
+        )
+
+        static let primaryDescription = NSLocalizedString(
+            "migration.welcome.primaryDescription",
+            value: "It looks like you’re switching from the WordPress app.",
+            comment: "The primary description in the migration welcome screen"
+        )
+
+        static func secondaryDescription(plural: Bool) -> String {
+            let comment = "The secondary description in the migration welcome screen"
+            let siteWord = plural ? "sites" : "site"
+            let value = "We found your \(siteWord). Continue to transfer all your data and sign in to Jetpack automatically."
+            if plural {
+                return NSLocalizedString("migration.welcome.secondaryDescription.plural", value: value, comment: comment)
+            } else {
+                return NSLocalizedString("migration.welcome.secondaryDescription.singular", value: value, comment: comment)
+            }
+        }
+
+        static let primaryButtonTitle = NSLocalizedString(
+            "Continue",
```

Comment:
@twstokes The only reason is because a translation of "Continue" and "Need help?" already exists. And if we use a different localization key ( for example: `migration.welcome.continue` ) it would result in duplicated translation for the same word. The `Localizable.strings` would look like this:

```
"Continue" = "Continue";
"migration.welcome.continue" = "Continue";
```

Also, it seems this "Continue" is being used in different places in the app:

<img width="300" alt="CleanShot 2022-11-04 at 15 57 15@2x" src="https://user-images.githubusercontent.com/9609223/200006573-603c7c1e-33ae-4ede-9253-6498a2b099a4.png">

Let me know if I'm missing something 🙏

Reply:
Ok sounds good! And since they have the same context reuse the English string-based key. 👍